### PR TITLE
add user comment deletion

### LINKED ISSUE #233: User can delete their own comments
API issue is #280

We will show a delete link under all of a user's own comments. 

image

We will show a delete confirmation modal:

image

After the comment has been deleted, we will delete the content of the comment, and show the text "[comment deleted by author]" or "[comment deleted by moderator]". The comment will stay in the comment tree. The user name and avatar will no longer be shown, nor will the upvoting arrows or the score. There is no way to un-delete a comment.

image

image

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #233 

#### What's this PR do?

this adds deletion functionality for the user's own comments.

#### How should this be manually tested?

Basically, if you're looking at your own comments you should see a 'delete' button. Clicking the button should open a confirmation dialog. If you click 'yes' the comment should update to look like this in the UI:

fjfjfjasdfasdfasdf

If you refresh the page you should see one of two things: if the comment you deleted did not have any replies it should just disappear, and if the comment in question had at least one reply you should see something like the above picture, with the replies sticking around under a 'deleted' placeholder.

If you're looking at someone else's comments you shouldn't see the 'delete' button.

### CODE REVIEW

File: static/js/components/CommentTree_test.js [hunk 1]
```diff
@@ -111,15 +114,35 @@ describe("CommentTree", () => {
   })
 
   it('should include an "Edit" button, if the user wrote the comment', () => {
-    comments[0].author_id = "username"
-    SETTINGS.username = "username"
+    // comments[0].author_id = "username"
```
Comment (noisecapella) [inline static/js/components/CommentTree_test.js]: Can you remove these lines?

File: static/js/containers/PostPage.js [hunk 2]
```diff
@@ -202,6 +202,19 @@ class PostPage extends React.Component<*, void> {
     )
   }
 
+  deleteComment = async () => {
+    // ⚠️  this is a destructive action! ⚠️
```
Comment (noisecapella) [inline static/js/containers/PostPage.js]: ⚠️

**Comment by noisecapella** [comment]
After the comment is deleted I see a 'delete' and 'edit' link on my comment:

screenshot from 2017-12-19 11-16-41

**Comment by noisecapella** [comment]
When I refresh the browser the comment doesn't have those two links anymore. However the 'remove' should probably also not be visible for a deleted comment

Other than this everything looks good

**Comment by aliceriot** [comment]
ok I made a small change to clear the `author_id` so that the 'delete' and 'edit' actions don't show up anymore.

The 'remove' action will only show up for moderators - do you think it's a big deal if they see that for a second? We don't have a clean way to tell whether a comment has been previously deleted or not, so I'm not sure of a good way to tell whether we want to show the 'remove' button or not. It's a small edge case anyway, since moderators likely will not be making lots of comments, I imagine.

File: static/js/components/CommentTree_test.js (hunk 1 above)
Comment (aliceriot) [reply static/js/components/CommentTree_test.js]: oopsie

**Comment by noisecapella** [comment]
That's fine with me

**Pull request merged**